### PR TITLE
Fail build if test run includes skipped tests without a developer provided reason

### PR DESCRIPTION
**Pull request by JoshuaGleason**
[[Link to Jira ticket](https://strongmind.atlassian.net/browse/DEVOPS-13059)](https://strongmind.atlassian.net/browse/TEAM-NUMBER)

## Purpose 

Fail build if test run includes skipped tests without a developer provided reason
## Approach 

Add action to output the rspec results to a json file.
Add action to parse the results and return a failing exit code if skipped tests match a specific pattern.
## Testing

Tested on a branch using repo dashboard
## Screenshots/Video

With skipped tests:
image

No tests skipped:
image

Still outputting test results to console:
image

### CODE REVIEW

File: .github/workflows/rspec-test.yml [hunk 1]
```diff
@@ -67,10 +67,20 @@ jobs:
               npm run test
           fi
 
-      - name: Test
-        if: ${{ steps.secret-check.outputs.available != 'true' }}
+      - name: Run RSpec to Output File
```
Comment (tpaulshippy) [inline .github/workflows/rspec-test.yml:70]: Slight problem here. Most of our projects have the secret to send code coverage to codeclimate. By removing that check, we will be running our tests twice, which will add quite a bit of time to the execution of this action.

File: .github/workflows/rspec-test.yml [hunk 2]
```diff
@@ -67,10 +67,20 @@ jobs:
...
+        id: rspec_test
         run: |
-          bundle exec rspec
+          bundle exec rspec --format json --out ${{ github.workspace }}/rspec_output.json
```
Comment (tpaulshippy) [inline .github/workflows/rspec-test.yml:73]: Can we not read the output of the previous step, without having to explicitly output to a json file?

File: scripts/parse_rspec_json_output.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    specific_pending_messages = [
+        "Temporarily skipped with xit",
+        "Not yet implemented",
+        "Add a hash of attributes"
+    ]
+
+    skipped_tests = [
+        example for example in data['examples']
+        if example['status'] == 'pending' and
+           any(msg in example.get('pending_message', '') for msg in specific_pending_messages)
```
Comment (tpaulshippy) [inline scripts/parse_rspec_json_output.py:17]: This is more elegant than just string matching on the output. I do like that.

File: .github/workflows/rspec-test.yml [hunk 4]
```diff
@@ -67,10 +67,20 @@ jobs:
...
+          
+      - name: Checkout public-reusable
+        uses: actions/checkout@v4
+        with:
+          repository: 'Strongmind/public-reusable-workflows'
```
Comment (tpaulshippy) [inline .github/workflows/rspec-test.yml:78]: Pulling down the whole repo just to run a script seems...smelly. Isn't there a way to download a single file from a public github repo?

**Review by tpaulshippy — CHANGES_REQUESTED** [review]

File: .github/workflows/rspec-test.yml (hunk 1 above)
Comment (tpaulshippy) [reply .github/workflows/rspec-test.yml:70]: I suppose we could split this out into another workflow entirely and have it run in parallel? Not in love with the duplication of that though.

File: .github/workflows/rspec-test.yml (hunk 2 above)
Comment (tpaulshippy) [reply .github/workflows/rspec-test.yml:73]: I think we need to add `--format documentation` before `--format json` to still get our output (see https://til.hashrocket.com/posts/8d0e6c5ede-rspec-multiple-formats-with-different-out-buffers)

Comment (JoshuaGleason) [reply .github/workflows/rspec-test.yml:73]: research indicated the $GITHUB_OUTPUT was best used with variables and short strings and would not be best used with content from the logs

Comment (JoshuaGleason) [reply .github/workflows/rspec-test.yml]: Done, added new commits and screenshot.

File: .github/workflows/rspec-test.yml (hunk 1 above)
Comment (JoshuaGleason) [reply .github/workflows/rspec-test.yml]: Added test back, no longer running rspec twice

**Review by tpaulshippy — APPROVED** [review]

**Pull request merged**